File: 238_product_of_array_except_self/238_product_of_array_except_self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        items = [1] * len(nums)

        prefix = 1
        for i in range(len(nums)):
            items[i] = prefix
            prefix = prefix * nums[i]
        
        postfix = 1
        for i in range(len(nums)-1, -1, -1):
            items[i] = items[i] * postfix
            postfix = postfix * nums[i]
        return items

File: 238_product_of_array_except_self/238_product_of_array_except_self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    let items = []

    let prefix = 1
    let postfix = 1

    for(let i=0; i< nums.length; i++) {
        items[i] = prefix;
        prefix = prefix * nums[i];
    }

    for(let i=nums.length-1 ; i>= 0; i--) {
        items[i] = items[i] * postfix;
        postfix = postfix * nums[i];
    }
    return items
};
